homeassistant/components/motion_blinds/config_flow.py

```diff
@@ -31,33 +37,71 @@ def __init__(self):
         """Initialize the Motion Blinds flow."""
         self.host = None
         self.key = None
+        self.ips = []
 
     async def async_step_user(self, user_input=None):
         """Handle a flow initialized by the user."""
         errors = {}
         if user_input is not None:
-            self.host = user_input[CONF_HOST]
-            self.key = user_input[CONF_API_KEY]
-            return await self.async_step_connect()
+            self.host = user_input.get(CONF_HOST)
+
+            if self.host is not None:
+                return await self.async_step_connect()
+
+            # Use MotionGateway discovery
+            discover_class = MotionDiscovery()
+            gateways = await self.hass.async_add_executor_job(discover_class.discover)
+            self.ips = list(gateways.keys())
+
+            if len(self.ips) == 1:
+                self.host = self.ips[0]
+                return await self.async_step_connect()
+
+            if len(self.ips) > 1:
+                return await self.async_step_select()
+
+            errors["base"] = "discovery_error"
 
         return self.async_show_form(
             step_id="user", data_schema=CONFIG_SCHEMA, errors=errors
         )
 
+    async def async_step_select(self, user_input=None):
+        """Handle multiple motion gateways found."""
+        errors = {}
+        if user_input is not None:
+            self.host = user_input["select_ip"]
+            return await self.async_step_connect()
+
+        select_schema = vol.Schema({vol.Required("select_ip"): vol.In(self.ips)})
+
+        return self.async_show_form(
+            step_id="select", data_schema=select_schema, errors=errors
+        )
+
     async def async_step_connect(self, user_input=None):
         """Connect to the Motion Gateway."""
+        errors = {}
+        if user_input is not None:
+            self.key = user_input[CONF_API_KEY]
+
+            connect_gateway_class = ConnectMotionGateway(self.hass, None)
```

Comment:
I could see `multicast=False` being slightly more clear

Reply:
Thanks for the tip.
I made it: `ConnectMotionGateway(self.hass, multicast=None)`
Since `False` will result in an error in the upstream lib.